recipes/configure_server.rb

```diff
@@ -136,12 +136,12 @@
 end
 
 # now let's set the root password only if this is the initial install
-unless node["percona"]["skip_passwords"]
+unless node["percona"]["skip_passwords"] || passwords.root_password.to_s.empty?
   root_pw = passwords.root_password
 
   execute "Update MySQL root password" do  # ~FC009 - `sensitive`
     command "mysqladmin --user=root --password='' password '#{root_pw}'"
-    only_if "mysqladmin --user=root --password='' version"
+    not_if "mysqladmin --user=root --password='#{root_pw}' version"
```

Comment:
Well it didn't work before too. The comment above `# now let's set the root password only if this is the initial install`